### ISSUE TITLE
Test suite failures when Sub::Util isn't present

### ISSUE THREAD

**Issue by pghmcfc** [houseabsolute/DateTime.pm#77]
```
$ make test
PERL_DL_NONLAZY=1 /usr/bin/perl "-MExtUtils::Command::MM" "-e" "test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
# 
# Versions for all modules listed in MYMETA.json (including optional ones):
# 
# === Configure Requires ===
# 
#     Module               Want Have
#     -------------------- ---- ----
#     Dist::CheckConflicts 0.02 0.11
#     ExtUtils::MakeMaker   any 6.68
# 
# === Configure Suggests ===
# 
#     Module      Want    Have
#     -------- ------- -------
#     JSON::PP 2.27300 2.97001
# 
# === Build Requires ===
# 
#     Module              Want Have
#     ------------------- ---- ----
#     ExtUtils::MakeMaker  any 6.68
# 
# === Test Requires ===
# 
#     Module                    Want     Have
#     ------------------------ ----- --------
#     CPAN::Meta::Check        0.011    0.014
#     CPAN::Meta::Requirements   any    2.140
#     ExtUtils::MakeMaker        any     6.68
#     File::Spec                 any     3.40
#     Storable                   any     2.45
#     Test::Fatal                any    0.014
#     Test::More                0.96 1.302133
#     Test::Warnings           0.005    0.026
#     utf8                       any     1.09
# 
# === Test Recommends ===
# 
#     Module         Want     Have
#     ---------- -------- --------
#     CPAN::Meta 2.120900 2.150010
# 
# === Runtime Requires ===
# 
#     Module                     Want  Have
#     -------------------------- ---- -----
#     Carp                        any  1.26
#     DateTime::Locale           1.06  1.17
#     DateTime::TimeZone         2.02  2.17
#     Dist::CheckConflicts       0.02  0.11
#     POSIX                       any  1.30
#     Params::ValidationCompiler 0.26  0.27
#     Scalar::Util                any  1.27
#     Specio                     0.18  0.42
#     Specio::Declare             any  0.42
#     Specio::Exporter            any  0.42
#     Specio::Library::Builtins   any  0.42
#     Specio::Library::Numeric    any  0.42
#     Specio::Library::String     any  0.42
#     Try::Tiny                   any  0.30
#     XSLoader                    any  0.16
#     base                        any  2.18
#     integer                     any  1.00
#     namespace::autoclean       0.19  0.28
#     overload                    any  1.18
#     parent                      any 0.236
#     strict                      any  1.07
#     warnings                    any  1.13
#     warnings::register          any  1.02
# 
t/00-report-prereqs.t .... ok
#   Failed test 'use DateTime;'
#   at t/00load.t line 6.
#     Tried to use 'DateTime'.
#     Error:  Cannot name a generated validation subroutine. Please install Sub::Util. at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime/Duration.pm line 276.
# Compilation failed in require at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
# BEGIN failed--compilation aborted at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
# Compilation failed in require at t/00load.t line 6.
# BEGIN failed--compilation aborted at t/00load.t line 6.
# Looks like you failed 1 test of 1.
t/00load.t ............... 
Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/1 subtests 
Cannot name a generated validation subroutine. Please install Sub::Util. at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime/Duration.pm line 276.
Compilation failed in require at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
BEGIN failed--compilation aborted at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
Compilation failed in require at t/01sanity.t line 6.
BEGIN failed--compilation aborted at t/01sanity.t line 6.
t/01sanity.t ............. 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
Cannot name a generated validation subroutine. Please install Sub::Util. at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime/Duration.pm line 276.
Compilation failed in require at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
BEGIN failed--compilation aborted at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
Compilation failed in require at t/02last-day.t line 7.
BEGIN failed--compilation aborted at t/02last-day.t line 7.
t/02last-day.t ........... 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
Cannot name a generated validation subroutine. Please install Sub::Util. at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime/Duration.pm line 276.
Compilation failed in require at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
BEGIN failed--compilation aborted at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
Compilation failed in require at t/03components.t line 6.
BEGIN failed--compilation aborted at t/03components.t line 6.
t/03components.t ......... 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
Cannot name a generated validation subroutine. Please install Sub::Util. at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime/Duration.pm line 276.
Compilation failed in require at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
BEGIN failed--compilation aborted at /builddir/build/BUILD/DateTime-1.47/blib/lib/DateTime.pm line 14.
Compilation failed in require at t/04epoch.t line 7.
BEGIN failed--compilation aborted at t/04epoch.t line 7.
t/04epoch.t .............. 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
...
t/48rt-115983.t .......... 
Dubious, test returned 2 (wstat 512, 0x200)
No subtests run 
Warning: DateTime::Format::Strptime did not compile at /usr/share/perl5/vendor_perl/Dist/CheckConflicts.pm line 184.
# Conflicts detected for DateTime:
#   DateTime::Format::Strptime is version unknown, but must be greater than version 1.1000
t/zzz-check-breaks.t ..... ok
Test Summary Report
-------------------
t/00load.t             (Wstat: 256 Tests: 1 Failed: 1)
  Failed test:  1
  Non-zero exit status: 1
...
t/48rt-115983.t        (Wstat: 512 Tests: 0 Failed: 0)
  Non-zero exit status: 2
  Parse errors: No plan found in TAP output
Files=50, Tests=6,  5 wallclock secs ( 0.06 usr  0.03 sys +  4.05 cusr  0.45 csys =  4.59 CPU)
Result: FAIL
Failed 48/50 test programs. 3/6 subtests failed.
make: *** [test_dynamic] Error 255
```

**Comment by pghmcfc** [comment]
#78 fixes this for me.

**Comment by autarch** [comment]
Fixed in 1.48.
